src/QsCompiler/Compiler/Templates/CppInterop.cs

```diff
@@ -0,0 +1,168 @@
+                {
+                    DataType.BoolType => "vector<char>",
+                    DataType.IntegerType => "vector<int64_t>",
+                    DataType.DoubleType => "vector<double_t>",
+                    DataType.PauliType => "std::vector<PauliId>",
+                    DataType.RangeType => "vector<RangeTuple>",
+                    DataType.ResultType => "vector<char>",
+                    DataType.StringType => "vector<string>",
+                },
+            };
+        }
+
+        public string? CppVarInitialValue()
+        {
+            return this.Type switch
+            {
+                DataType.BoolType => "InteropFalseAsChar",
+                DataType.IntegerType => "0",
+                DataType.DoubleType => "0.0",
+                DataType.PauliType => "PauliId::PauliId_I",
+                DataType.RangeType => null,
+                DataType.ResultType => "InteropResultZeroAsChar",
+                DataType.StringType => null,
+                DataType.ArrayType => this.ArrayType switch
+                {
+                    DataType.BoolType => null,
+                    DataType.IntegerType => null,
+                    DataType.DoubleType => null,
+                    DataType.PauliType => null,
+                    DataType.RangeType => null,
+                    DataType.ResultType => null,
+                    DataType.StringType => null,
+                },
+            };
+        }
+
+        public string? TransformationType()
+        {
+            return this.Type switch
+            {
+                DataType.BoolType => "BoolAsCharMap",
+                DataType.IntegerType => null,
+                DataType.DoubleType => null,
+                DataType.PauliType => "PauliMap",
+                DataType.RangeType => null,
+                DataType.ResultType => "ResultAsCharMap",
+                DataType.StringType => null,
+                DataType.ArrayType => this.ArrayType switch
+                {
+                    DataType.BoolType => "BoolAsCharMap",
+                    DataType.IntegerType => null,
+                    DataType.DoubleType => null,
+                    DataType.PauliType => "PauliMap",
+                    DataType.RangeType => null,
+                    DataType.ResultType => "ResultAsCharMap",
+                    DataType.StringType => null,
+                },
+            };
+        }
+    }
+
+    public class EntryPointOperationCpp : EntryPointOperation
+    {
+        public List<ArgumentCpp> InteropArguments;
+
+        public EntryPointOperationCpp(EntryPointOperation entryPointOperation)
+        {
+            this.Name = entryPointOperation.Name;
+            this.Arguments = entryPointOperation.Arguments.ToList();
+            this.Arguments.Sort((a, b) => a.Position.CompareTo(b.Position));
+            this.InteropArguments = entryPointOperation.Arguments.Select(arg => new ArgumentCpp(arg)).ToList();
```

Comment:
```suggestion
            this.InteropArguments = this.Arguments.Select(arg => new ArgumentCpp(arg)).ToList();
```

Reply:
This is to make sure that InteropArguments are sorted.

Reply:
good catch! fixed